### DISCUSSION TITLE
Lambda response payload size being exceeded

### DISCUSSION THREAD

**Discussion by james-astra** [spatie/laravel-pdf#118] [Q&A] [answered]
### What happened?

Hello I am using onLambda() to generate my PDF with sidecar which is all okay except I'm wondering if it's possible to run both on lambda and also save directly to S3 within Lambda? I know that this is possible in Browsershot however I can't work out how to achieve this with this pdf package.

I'm exceeding the response payload size however I don't want to have it respond the file, just save it straight to S3.

Please could someone advise on how I can achieve this?

### How to reproduce the bug

Create a PDF with x amount of content, run on lambda and try to save to S3 using ->disk()

### Package Version

1.4

### PHP Version

8.2

### Laravel Version

10.0

### Which operating systems does with happen with?

Linux

**Comment by james-astra** [comment]
Hi, I forked the repository and made a quick & dirty usage of the feature in browser shot. If anyone needs a quick solution it's here: https://github.com/astra-telematics/laravel-pdf/tree/save-to-s3-in-lambda

**Comment by rvzug** [comment]
I'm wondering if @freekmurze would accept a PR for this? I could also use a stable way to save pdfs directly to s3. Or what could make it work for you?

**Comment by james-astra** [comment]
> I'm wondering if @freekmurze would accept a PR for this? I could also use a stable way to save pdfs directly to s3. Or what could make it work for you?

If you check out my fork linked above you will be able to see how I got it to work. It's dirty and not ready for a PR but I'm happy to implement it properly for PR if one would be accepted

**Reply by rvzug** [reply]
Well, I will first try [this](https://github.com/stefanzweifel/sidecar-browsershot?tab=readme-ov-file#saving-directly-to-s3). But if some things can be added to this package, let's help the community ;-)

**Reply by james-astra** [reply, marked as answer]
This package uses browsershot :) I would recommend using this package and use my fork if you would like to save directly to S3 from your lambda. Here is how you can use it:
```php
$pdf = Pdf::view('view.name', $data);
$pdf->withBrowsershot(function (Browsershot $browsershot) {
    $browsershot->setOption('args', ['--disable-web-security']);
    $browsershot->waitUntilNetworkIdle();
});
$pdf->onLambda()->disk('s3-bucket')->save('example.pdf');
```

**Comment by Gandhi11** [comment]
@freekmurze @james-astra I am facing the same problem, is there any way to get this as part of the default package?

**Reply by PymPiDev** [reply]
Yes, using laravel-pdf you will do it this way

```php
$pdf = \Spatie\LaravelPdf\Facades\Pdf::view('view.name', $data);
            
$pdf->onLambda()->getBrowsershot()->saveToS3('path/example.pdf', 'disk.name');
```
